portfolio/src/main/java/com/google/sps/servlets/DropoffServlet.java

```diff
@@ -0,0 +1,46 @@
+package com.google.sps.servlets;
+
+import com.google.sps.data.ScrapDropoff;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+// Will send JSON formatted coordinates of dropoff locations to 
+// script.js when fetched
+@WebServlet("/dropoff-data")
+public class DropoffServlet extends HttpServlet {
+  
+  private Collection<ScrapDropoff> dropoffs;
+  
+  // Create array list with parse csv values as ScrapDropoff objects
+  @Override
+  public void init(){
+    dropoffs = new ArrayList<>();
+    Scanner scanner = new Scanner(getServletContext().getResourceAsStream("/WEB-INF/dropoff-locations.csv"));
+    while (scanner.hasNextLine()) {
+      String line = scanner.nextLine();
+      String[] cells = line.split(",");
+
+      double lat = Double.parseDouble(cells[0]);
```

Comment:
This looks like a good use case for @Autovalue, or a helper class  if you're interested in trying it out. It will help keep these array indexes named, and distinguishable. One obvious pitfall is mixing up the indices = setting lat = cell[1] and lon = cells[0]. With a helper object this can become lat = cells.lat etc.

Reply:
I looked into autovalue, and to me it seemed like a helper class made more sense here because I can feed in the full line and parse in the constructor of the class. In what I read about autovalue, I wasn't sure how I could do that since the compiler creates so much of the functionality at runtime. I added a helper class implementation (Coordinate.java), but let me know if you think that autovalue would be better suited here and if there is a good place to start with that.